include/zephyr/arch/arc/arch.h

```diff
@@ -344,6 +344,10 @@ static ALWAYS_INLINE void arch_nop(void)
 	__builtin_arc_nop();
 }
 
+#ifdef CONFIG_XIP
+extern char __arc_rw_sram_size[];
+#endif /* CONFIG_XIP */
```

Comment:
We have this definition if CONFIG_XIP is enabled but we use and set it if XIP is disabled.

Looks like a bug.

Reply:
Fixed